pkg/services/validator/azure.go

```diff
@@ -223,82 +231,62 @@ func configureAzureRBACRules(c *components.AzurePluginConfig) error {
 	if !addRules {
 		return nil
 	}
-
-	log.InfoCLI("Note: You must configure at least one rule for plugin configuration.")
-	ruleIdx := len(c.Validator.RBACRules)
-
 	for {
-		log.InfoCLI("Note: Collecting input for rule #%d", ruleIdx+1)
-
-		// This is intentional. We only have one rule type in the Azure plugin now, so we don't
-		// prompt the user for rule type. But we are keeping the rest of the type-oriented code here
-		// (e.g. the rule types tracked in the YAML config file) to avoid less refactor work later
-		// when we add more rule types for Azure.
-		ruleType := ruleTypeRBAC
-		c.RuleTypes[ruleIdx] = ruleType
-
-		rule := &plug.RBACRule{
-			Permissions: []plug.PermissionSet{},
-		}
-
-		switch ruleType {
-		case ruleTypeRBAC:
-			if err := configureAzureRBACRule(&ruleNames, rule); err != nil {
-				return fmt.Errorf("failed to configure RBAC rule: %w", err)
-			}
-		default:
-			return fmt.Errorf("unknown rule type (%s)", ruleType)
+		if err := readRBACRule(c, nil, -1, ruleNames); err != nil {
+			return err
 		}
-
-		c.Validator.RBACRules = append(c.Validator.RBACRules, *rule)
-
-		addRBACRule, err := prompts.ReadBool("Add additional RBAC rule", false)
+		add, err := prompts.ReadBool("Add additional RBAC rule", false)
 		if err != nil {
-			return fmt.Errorf("failed to prompt for bool for add an RBAC rule: %w", err)
+			return err
 		}
-		if !addRBACRule {
+		if !add {
 			break
 		}
-		ruleIdx++
 	}
-
 	return nil
 }
 
-func initRbacRule(ruleNames *[]string, r *plug.RBACRule) error {
-	var err error
-	if r.Name != "" {
-		log.InfoCLI("Reconfiguring RBAC rule: %s", r.Name)
-		*ruleNames = append(*ruleNames, r.Name)
+// readRBACRule begins the process of reconfiguring or beginning a new RBAC rule.
+func readRBACRule(c *components.AzurePluginConfig, r *plug.RBACRule, idx int, ruleNames *[]string) error {
+	if r == nil {
+		r = &plug.RBACRule{}
+	}
+
+	name := r.Name
```

Comment:
Please refactor the Azure rule initialization to use the `initRule` pattern used for other plugins, e.g., AWS.

Reply:
That's my preference - in the spirit of writing DRY code. Yeah, reflection is a bit ugly, but I'd rather use it than see those ~10 lines of name initialization code duplicated for every rule type.